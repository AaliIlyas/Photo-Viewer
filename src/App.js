import './App.css';
import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { RenderPage } from './photoViewer/renderPage';

function App() {
	return (
		<Router>
			<Switch>
				<Route exact path="/">
					<RenderPage />
				</Route>
				<Route exact path="/images/:id" >
					<RenderPage />
				</Route>
				<Route path="/">
					<div>
						<h1>Error: 404</h1>
					</div>
				</Route>
			</Switch>
		</Router>
	);
}

export default App;
