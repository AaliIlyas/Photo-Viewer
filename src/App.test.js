import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import ImageUrls from './photoViewer/imageSelector';
import renderer from 'react-test-renderer';

test('renders the Gallery text', () => {
	render(<App />);
	const text = screen.getByText(/Gallery/i);
	expect(text).toBeInTheDocument();
});

test('check 1st image url is correct', () => {
	render(<App />);
	const image = screen.getByTestId("main-img");

	expect(image).toHaveAttribute('src', ImageUrls[0])
});

//A ‘Regression / Snapshot' test to confirm that the ‘ImageViewer' component doesn’t accidentally change in the future.

describe("Snapshot testing:", () => {
	it('site renders correctly', () => {
		const app = renderer.create(<App />).toJSON();

		expect(app).toMatchSnapshot();
	});
});


test('site renders 44 thumbnail images', () => {
	render(<App />);
	const images = screen.getAllByRole("thumbnail");
	expect(images).toHaveLength(44);

})

test('check that thumbnail correctly updates main image', () => {
	render(<App />);
	const image = screen.getByTestId("main-img");

	userEvent.click(screen.getByTestId('3'));

	expect(image).toHaveAttribute('src', ImageUrls[3]);

});
