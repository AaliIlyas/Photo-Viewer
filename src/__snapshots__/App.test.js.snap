// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing: site renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    React Photo Viewer
  </h1>
  <div
    className="main"
  >
    <img
      className="main-img"
      data-testid="main-img"
      id="main-img"
      src="https://picsum.photos/id/600/1600/900.jpg"
    />
  </div>
  <h2>
    -- Gallery --
  </h2>
  <div>
    <img
      className="thumbnail"
      data-testid={0}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/600/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={1}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/602/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={2}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/603/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={3}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/604/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={4}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/605/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={5}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/606/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={6}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/607/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={7}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/608/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={8}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/609/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={9}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/610/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={10}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/611/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={11}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/612/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={12}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/613/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={13}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/614/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={14}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/615/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={15}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/616/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={16}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/617/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={17}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/618/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={18}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/619/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={19}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/620/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={20}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/621/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={21}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/622/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={22}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/623/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={23}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/625/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={24}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/626/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={25}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/627/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={26}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/628/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={27}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/629/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={28}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/630/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={29}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/631/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={30}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/633/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={31}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/634/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={32}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/635/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={33}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/637/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={34}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/638/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={35}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/639/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={36}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/640/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={37}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/641/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={38}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/642/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={39}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/643/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={40}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/645/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={41}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/646/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={42}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/648/1600/900.jpg"
    />
    <img
      className="thumbnail"
      data-testid={43}
      onClick={[Function]}
      role="thumbnail"
      src="https://picsum.photos/id/649/1600/900.jpg"
    />
  </div>
</div>
`;
